src/main/java/gift/validation/ProductNameValidator.java

```diff
@@ -0,0 +1,17 @@
+package gift.validation;
+
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
+
+public class ProductNameValidator implements ConstraintValidator<ValidProductName, String> {
+
+    private static final String ALLOWED_SPECIAL_CHARS = "()+\\-\\[\\]&/_";
+    private static final String PATTERN = "^[a-zA-Z0-9" + ALLOWED_SPECIAL_CHARS + "가-힣\\s]*$";
```

Comment:
상수로 의미 구분해서 가독성이 좋아졌네요 👍
참고로 저는 이렇게 바로 쉽게 읽히지 않는 부분들은 주석을 활용해서 가독성을 조금이라도 높이는 걸 선호합니다 ㅎㅎ
```suggestion
    private static final String ALLOWED_SPECIAL_CHARS = "()+\\-\\[\\]&/_"; // ( ), [ ], +, -, &, /, _
    private static final String PATTERN = "^[a-zA-Z0-9" + ALLOWED_SPECIAL_CHARS + "가-힣\\s]*$";
```